[ariel-os-sensors] cannot compile with max-sample-min-count > 1

### Summary

```
cargo clippy --config ariel-os-cargo.toml -p ariel-os-sensors
```
Does not return any error, but running 

```
cargo clippy --config ariel-os-cargo.toml -p ariel-os-sensors --features "max-sample-min-count-2"
```

Throws an error, looks like there's an issue in the macro, with how the iterators are created in `Samples::samples()` and `ReadingChannels::iter()`.

### Relevant log output

```shell
error[E0308]: `match` arms have incompatible types
  --> src/ariel-os-sensors/src/sensor.rs:17:1
   |
17 | ariel_os_macros::define_count_adjusted_sensor_enums!();
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected `1`, found `2`
   | this is found to be of type `core::array::IntoIter<sample::Sample, 1>`
   | `match` arms have incompatible types
   |
   = note: expected struct `core::array::IntoIter<_, 1>`
              found struct `core::array::IntoIter<_, 2>`
```

### Hardware/ Setup

_No response_

### Possible Solution

_No response_

### Would you like to work on fixing this bug?

Yes

--- a/src/ariel-os-macros/src/define_count_adjusted_sensor_enums.rs
+++ b/src/ariel-os-macros/src/define_count_adjusted_sensor_enums.rs
@@ -67,7 +67,7 @@ pub fn define_count_adjusted_sensor_enums(_item: TokenStream) -> TokenStream {
     let reading_channels_iter = (1..=count)
         .map(|i| {
             let variant = variant_name(i);
-            quote! { InnerReadingChannels::#variant(ref samples) => samples.iter().copied() }
+            quote! { InnerReadingChannels::#variant(ref channels) => channels.iter().copied() }
         });
 
     let expanded = quote! {

